apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java

```diff
@@ -165,20 +173,32 @@ public List<NamespaceBO> findNamespaceBOs(String appId, Env env, String clusterN
       throw new BadRequestException("namespaces not exist");
     }
 
-    List<NamespaceBO> namespaceBOs = new LinkedList<>();
+    List<NamespaceBO> namespaceBOs = new CopyOnWriteArrayList<>();
+    CountDownLatch latch = new CountDownLatch(namespaces.size());
     for (NamespaceDTO namespace : namespaces) {
+      executorService.submit(() -> {
+        NamespaceBO namespaceBO;
+        try {
+          namespaceBO = transformNamespace2BO(env, namespace);
+          namespaceBOs.add(namespaceBO);
+        } catch (Exception e) {
+          LOGGER.error("parse namespace error. app id:{}, env:{}, clusterName:{}, namespace:{}",
+              appId, env, clusterName, namespace.getNamespaceName(), e);
+        } finally {
+          latch.countDown();
+        }
+      });
 
-      NamespaceBO namespaceBO;
-      try {
-        namespaceBO = transformNamespace2BO(env, namespace);
-        namespaceBOs.add(namespaceBO);
-      } catch (Exception e) {
-        LOGGER.error("parse namespace error. app id:{}, env:{}, clusterName:{}, namespace:{}",
-            appId, env, clusterName, namespace.getNamespaceName(), e);
-        throw e;
-      }
+    }
+    try {
+      latch.await();
+    } catch (InterruptedException e) {
+      //ignore
     }
 
+    if(namespaceBOs.size() != namespaces.size()){
+       throw new RuntimeException("Parse namespace error");
```

Comment:
How about 

* encode size to error message
* find which namespace cannot get

for example

```suggestion
       throw new RuntimeException("Parse namespaces error, expect {} but get {}, cannot get those namespaces: {}");
```
This will give more information when trouble shoot.

Reply:
This is a good idea